tests/vmi_tpm_test.go

```diff
@@ -85,3 +101,182 @@ var _ = Describe("[sig-compute]vTPM", decorators.SigCompute, func() {
 		})
 	})
 })
+
+var _ = Describe("[sig-storage]vTPM", func() {
+	var virtClient kubecli.KubevirtClient
+	var err error
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("with persistent TPM VM option enabled", func() {
+		restartVM := func(vm *v1.VirtualMachine) {
+			By("Stopping the VM")
+			err = virtClient.VirtualMachine(vm.Namespace).Stop(context.Background(), vm.Name, &v1.StopOptions{})
+			ExpectWithOffset(1, err).ToNot(HaveOccurred())
+			EventuallyWithOffset(1, func() error {
+				_, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+				return err
+			}, 300*time.Second, 1*time.Second).ShouldNot(Succeed())
+
+			By("Starting the VM")
+			err = virtClient.VirtualMachine(vm.Namespace).Start(context.Background(), vm.Name, &v1.StartOptions{})
+			ExpectWithOffset(1, err).ToNot(HaveOccurred())
+			var vmi *v1.VirtualMachineInstance
+			EventuallyWithOffset(1, func() error {
+				vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+				return err
+			}, 300*time.Second, 1*time.Second).Should(Succeed())
+			libwait.WaitForSuccessfulVMIStartWithTimeout(vmi, 60)
+
+			By("Logging in as root")
+			err = console.LoginToFedora(vmi)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		migrateVMI := func(vmi *v1.VirtualMachineInstance) {
+			By("Migrating the VMI")
+			checks.SkipIfMigrationIsNotPossible()
+			migration := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+			tests.RunMigrationAndExpectCompletion(virtClient, migration, tests.MigrationWaitTime)
+
+		}
+
+		checkTPM := func(vmi *v1.VirtualMachineInstance) {
+			By("Ensuring the TPM is still functional and its state carried over")
+			ExpectWithOffset(1, console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "tpm2_unseal -Q --object-context=0x81010002\n"},
+				&expect.BExp{R: "MYSECRET"},
+			}, 300)).To(Succeed(), "the state of the TPM did not persist")
+		}
+
+		DescribeTable("[Serial]should persist TPM secrets across", Serial, func(op1, op2 string) {
+			By("Setting the backend storage class to the default for RWX FS or skip if missing")
+			storageClass, exists := libstorage.GetRWXFileSystemStorageClass()
+			Expect(exists).To(BeTrue(), "No RWX FS storage class found")
```

Comment:
There is an inconsistent behavior between this `Describe` and the subsequent `It`:
Here we will fail if the storageClass does not exist, later we silently Skip the test.
Maybe this can converge in the same behavior inside the `BeforeEach` block.
Thanks

Reply:
In the `BeforeEach` you could also directly adjust the KubevirtCR (marking the whole `Context` as Serial), and the vmi itself